extensions/hibernate-search-backend-elasticsearch/deployment/src/main/java/io/quarkus/hibernate/search/backend/elasticsearch/deployment/HibernateSearchBackendElasticsearchProcessor.java

```diff
@@ -0,0 +1,153 @@
+package io.quarkus.hibernate.search.backend.elasticsearch.deployment;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.Set;
+
+import org.hibernate.search.backend.elasticsearch.analysis.ElasticsearchAnalysisConfigurer;
+import org.hibernate.search.backend.elasticsearch.gson.spi.GsonClasses;
+import org.hibernate.search.backend.elasticsearch.index.layout.IndexLayoutStrategy;
+
+import io.quarkus.arc.deployment.UnremovableBeanBuildItem;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.BuildSteps;
+import io.quarkus.deployment.builditem.ApplicationArchivesBuildItem;
+import io.quarkus.deployment.builditem.HotDeploymentWatchedFileBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.NativeImageResourceBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+import io.quarkus.hibernate.search.backend.elasticsearch.runtime.HibernateSearchBackendElasticsearchBuildTimeConfig;
+import io.quarkus.hibernate.search.backend.elasticsearch.runtime.MapperContext;
+import io.quarkus.runtime.configuration.ConfigurationException;
+
+@BuildSteps
+class HibernateSearchBackendElasticsearchProcessor {
+
+    @BuildStep
+    void registerBeans(List<HibernateSearchBackendElasticsearchEnabledBuildItem> searchEnabledPUs,
```

Comment:
```suggestion
    void registerBeans(List<HibernateSearchBackendElasticsearchEnabledBuildItem> elasticsearchEnabled,
```
a bit of nitpicking 🙈 feel free to ignore 😄